### ISSUE TITLE
How to use stopPainting() ?

### ISSUE THREAD

**Issue by zengming00** [electron/electron#11715]
* Electron version: 1.7.11
* Operating system: win10 x64

i want record every frame to make a video,

### Expected behavior

stop painting page, until call startPainting()

### Actual behavior

not stop painting page

### How to reproduce
```js
const electron = require('electron');
const fs = require('fs');
const app = electron.app
const BrowserWindow = electron.BrowserWindow

let mainWindow

// app.disableHardwareAcceleration()
function createWindow() {
  mainWindow = new BrowserWindow({
    width: 480, height: 800, frame: false, resizable: false, // show: false,
    webPreferences: {
      offscreen: true
    }
  })
  // mainWindow = new BrowserWindow({ width: 480, height: 800 })
  var i = 0;
  mainWindow.webContents.on('paint', (event, dirty, image) => {
    mainWindow.webContents.stopPainting()
    const data = image.toPNG()
    fs.writeFileSync('./f/' + i++ + '.png', data)
    // mainWindow.webContents.startPainting()
  })
  mainWindow.webContents.setFrameRate(1)

  mainWindow.on('closed', function () {
    mainWindow = null
  })
  
  var url = 'https://github.com';
  mainWindow.loadURL(url)
  //mainWindow.webContents.openDevTools()

}

app.on('ready', createWindow)
app.on('window-all-closed', function () {
  if (process.platform !== 'darwin') {
    app.quit()
  }
})
app.on('activate', function () {
  if (mainWindow === null) {
    createWindow()
  }
})
```

**Comment by brenca** [comment]
It looks like the check for `IsPainting` got lost during chromium updates in the GPU OSR path, you could use this with `app.disableHardwareAcceleration()` in the meantime.

**Comment by zengming00** [comment]
i tried app.disableHardwareAcceleration() ,can stop , but can not restart paint

**Comment by brenca** [comment]
I realized that calling `StartPainting` inside `OnPaint` causes stack overflow with `app.disableHardwareAcceleration()`, because it generates another frame (and thus calls `OnPaint` again). This, and the original problem should be fixed with #11728 and it's sister PRs for 1.8.x and 1.7.x

**Comment by codebytere** [comment]
Closing as this has been resolved by merging of the above PRs
